pkg/preflight/preflight.go

```diff
@@ -276,6 +299,20 @@ func CreateRenderedSpec(app *apptypes.App, sequence int64, origin string, inClus
 		}
 	}
 
+	tsKinds, err := kotsutil.LoadTSKindsFromPath(filepath.Join(archiveDir, "rendered"))
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to load troubleshoot kinds from path: %s", filepath.Join(archiveDir, "rendered")))
+	}
+
+	if tsKinds.PreflightsV1Beta2 != nil {
+		for _, v := range tsKinds.PreflightsV1Beta2 {
+			builtPreflight = troubleshootpreflight.ConcatPreflightSpec(builtPreflight, &v)
```

Comment:
doesn't this mean that collectors and analyzers from both preflight specs (the CRD and the secret) will be concatenated and executed? That doesn't seem to be the behavior for running preflights in the UI (the code [here](https://github.com/replicatedhq/kots/pull/3964/files#diff-e3a33423651aa73db11fec570357c6b99345bd78263668309b315244691765c4R85))

Reply:
ah, yeah. good catch. I think I had moved that out of the if/else to address one of your earlier comments, but I'll need to rearrange this again.